utils/install-app.js

```diff
@@ -141,26 +141,41 @@ function addToServer(appDir, installDir) {
   }
 }
 
-function loadRecognizers(appDir, appId, appVers) {
+function copyRecognizers(appDir, appId, appVers) {
   let recognizers;
+  let recognizersKeys;
   let configRecognizers;
   let configLocation;
 
   try { // Get recognizers in a plugin's appDir/config/xxx location
     recognizers = JSON.parse(fs.readFileSync(path.join(appDir, "config/recognizers", appId))).recognizers;
-    const keys = Object.keys(recognizers)
-    for (const key of keys) { // Add metadata for plugin version & plugin identifier of origin (though objects don't have to be plugin specific)
+    recognizersKeys = Object.keys(recognizers)
+    for (const key of recognizersKeys) { // Add metadata for plugin version & plugin identifier of origin (though objects don't have to be plugin specific)
       recognizers[key].pluginVersion = appVers;
       recognizers[key].pluginIdentifier = appId;
+      recognizers[key].key = appId + ":" + key + ":" + recognizers[key].id; // pluginid_that_provided_it:index(or_name)_in_that_provider:actionid
     }
     logger.debug('ZWED0294I', appId); //logger.debug("Found recognizers for '" + appId + "'");
   } catch (e) {
     logger.debug('ZWED0295I', (path.join(appDir, "config/recognizers", appId)), appId); //logger.debug("Could not find recognizers in '" + (path.join(appDir, "config/recognizers", appId)) + "'");
   }
 
+  if (process.env.INSTANCE_DIR) {
+    configLocation = path.join(process.env.INSTANCE_DIR, "workspace/app-server/ZLUX/pluginStorage/org.zowe.zlux.ng2desktop/");
+  } else {
+    configLocation = path.join(userInput.zluxConfig, "../../ZLUX/pluginStorage/org.zowe.zlux.ng2desktop/");
```

Comment:
zluxConfig should be read to find the value of the instance directory, rather than assuming it.
It would be `instanceDir`, meaning path.join(instanceDir, 'ZLUX/pluginStorage/org.zowe.zlux.ng2desktop/`);
see an example here https://github.com/zowe/zlux-app-server/blob/master/defaults/serverConfig/server.json#L43